rewrite-groovy/src/test/java/org/openrewrite/groovy/GroovyParserParentheseDiscovererTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.groovy;
+
+import lombok.SneakyThrows;
+import org.codehaus.groovy.ast.expr.Expression;
+import org.codehaus.groovy.ast.expr.MethodCallExpression;
+import org.junit.jupiter.api.Test;
+import sun.reflect.ReflectionFactory;
+
+import java.lang.reflect.Constructor;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class GroovyParserParentheseDiscovererTest {
+
+    @Test
+    void invoke() {
+        MethodCallExpression node = MockMethodCallExpression.of("invoke");
+
+        //language=groovy
+        String source = "(a.invoke())";
+        Integer result = GroovyParserParentheseDiscoverer.getInsideParenthesesLevel(node, source, 0);
+
+        assertEquals(1, result);
+    }
+
+    @Test
+    void invokeWithArguments() {
+        MethodCallExpression node = MockMethodCallExpression.of("invoke");
+
+        //language=groovy
+        String source = "(a.invoke(\"A\", \"\\$\", \"C?)\"))";
```

Comment:
If you're having to escape the double quotes then a text block might make more sense.
```suggestion
        String source = """
          (a.invoke("A", "\\$", "C?)"))
          """;
```